### ISSUE TITLE
maxDetectedFaces not working, what about fastMode?

### ISSUE THREAD

**Issue by kostadinnm** [WICG/shape-detection-api#39]
Hello.
I've been playing with the api. I am about to try to utilize it in a small desktop app, which is already a bit heavy. Although limiting the number of faces detected won't matter in my case, it came strange to me that this option doesn't work. Here are the results of the simple test I made:

**one face:**
![one_face](https://user-images.githubusercontent.com/18436405/37234970-a10d5fea-2403-11e8-99d4-44a1b56b4fd5.jpg)

**24 faces(all detected, although only 2 were requested):**
![24_faces](https://user-images.githubusercontent.com/18436405/37234972-a2706fe4-2403-11e8-8a34-748362982b24.jpg)

Actually, reading the draft kind of left me with the impression that those two options(along with `fastMode`) had never been implemented, but I just wanted to share it here, so that it's clear for anyone else :)
Always feel free to correct me!

Cheers,
Kostadin

**Comment by yellowdoge** [comment]
The actual implementation depends on the platform and its versions ATM :
- [Android core](https://cs.chromium.org/chromium/src/services/shape_detection/android/java/src/org/chromium/shape_detection/FaceDetectionImpl.java?q=numberOfFaces&sq=package:chromium&dr=C&l=79) and [Android GMS](https://cs.chromium.org/chromium/src/services/shape_detection/android/java/src/org/chromium/shape_detection/FaceDetectionImplGmsCore.java?sq=package:chromium&dr=C&l=72) both seem to have it wired at first glance. 
- [Mac Core Image](https://cs.chromium.org/chromium/src/services/shape_detection/face_detection_impl_mac.mm?sq=package:chromium&dr=C&l=16) seems to not support it or not have it wired but there is [support for it](https://developer.apple.com/documentation/coreimage/cidetectormaxfeaturecount?language=objc).  
- [Windows 10](https://cs.chromium.org/chromium/src/services/shape_detection/face_detection_provider_win.cc?sq=package:chromium&dr=C&l=43) seems to not use the max face count. 
- [Mac Vision](https://cs.chromium.org/chromium/src/services/shape_detection/face_detection_provider_mac.mm?type=cs&q=FaceDetectorOptionsPtr+file:%5Esrc/services/shape_detection/+package:%5Echromium$&l=29) is still WIP.

What's your platform? Can you use another?

**Comment by kostadinnm** [comment]
I plan to utilize it to enhance user activity detection within [stretcly](https://github.com/hovancik/stretchly), which is 
> a cross-platform electron app that reminds you to take breaks when working with computer

I'm on windows 10 now, but it'be great if it works in linux and windows 7 as well. For now, I'll go w/o passing any params to `FaceDetector`. Performance-wise, it'll be better if it allows me to opt for single face detection and fastMode, I guess.
I'd been looking at [tracking.js](https://github.com/eduardolundgren/tracking.js), then I realized it uses [clmtrackr.js](https://github.com/auduno/clmtrackr), which in turn uses [js-objectdetect](https://github.com/mtschirs/js-objectdetect)(correct me if I'm wrong), whose page has a comparison with [jsfeat](https://github.com/inspirit/jsfeat). The latter seemed inapplicable, but most recent. But then I found the shape detection api, which works surprising well and is keeping my believes on enhancing stretchly for now.

**Comment by yellowdoge** [comment]
> but it'be great if it works in linux and windows 7 as well.

I'm afraid that's beyond my control 😅 , the only thing we can do to ease web developers is to provide a polyfill (which I started and got somehow running for barcodes in this [repo](https://github.com/yellowdoge/shapedetection-polyfill)).

**Comment by yellowdoge** [comment]
Actually I was playing around with the Mac implementation that should support a maximum amount of faces via the [CIDetectorMaxFeatureCount](https://developer.apple.com/documentation/coreimage/cidetectormaxfeaturecount?language=objc) (re. my comment above): on a closer examination this is only available for Rectangle Detectors ([link](https://github.com/phracker/MacOSX-SDKs/blob/HEAD/MacOSX10.12.sdk/System/Library/Frameworks/CoreImage.framework/Versions/A/Headers/CIDetector.h#L110
)).

**Comment by kostadinnm** [comment]
Actually, I just now realized that electron is quite behind chromium release-cycle. It currently utilizes chromium 59, while the face detection `detect()` method seems to actually be working(in a small demo of mine) in a far later version - the latest chrome 65. The problem is that even electron-beta is sub- chromium 63, if I remember correctly.
I'm not much into how polyfills work, but I guess it's a wrapper that extends old frameworks with the newer apis, right?! If that is the case, I guess, there's some chance of having a polyfill for the face detection as well?! Otherwise, I'd have to use one of the aforementioned js libs, which is not how I imagined it :(
Cheers,
Kostadin
